src/main/resources/native/android/c/launcher.c

```diff
@@ -55,16 +55,22 @@ int argsize = 8;
 char** createArgs() {
 LOGE(stderr, "CREATE ARGS");
     int origSize = sizeof(origargs)/sizeof(char*);
-    char** result = malloc((origSize+1)* sizeof(char*));
+    char** result = malloc((origSize+2)* sizeof(char*));
     for (int i = 0; i < origSize; i++) {
-        result[i] = origargs[i];
+        result[i] = (char *)origargs[i];
     }
     int tmpArgSize=18+strnlen(appDataDir, 512);
     char* tmpArgs = calloc(sizeof(char), tmpArgSize);
     strcpy(tmpArgs,"-Djava.io.tmpdir=");
     strcat(tmpArgs,appDataDir);
     result[origSize]=tmpArgs;
     argsize++;
+    int userArgSize=13+strnlen(appDataDir, 512);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'userArgSize' is assigned a value that is never used.](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=5082599)